packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -905,9 +912,45 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
         }
       }
     }
+  }
+
+  /**
+   * Validate Performance Insights settings
+   */
+  private validatePerformanceInsightsSettings(instance: IAuroraClusterInstance): void {
+    // If Performance Insights is enabled on the cluster, the one for each instance will be enabled as well.
+    if (this.performanceInsightsEnabled && instance.performanceInsightsEnabled === false) {
+      Annotations.of(this).addWarningV2(
+        '@aws-cdk/aws-rds:instancePerformanceInsightsOverridden',
+        `Performance Insights is enabled on cluster '${this.node.id}' at cluster level, but disabled for instance '${instance.node.id}'. `+
+        'However, Performance Insights for this instance will also be automatically enabled if enabled at cluster level.',
+      );
+    }
 
-    if (performanceInsightsDuplicate) {
-      throw new Error('Cannot enable Performance Insights on both the cluster and the instances');
+    // If `performanceInsightRetention` is enabled on the cluster, the same parameter for each instance must be
+    // undefined or the same as the value at cluster level.
+    if (
+      this.performanceInsightRetention &&
+      instance.performanceInsightRetention &&
+      instance.performanceInsightRetention !== this.performanceInsightRetention
+    ) {
+      throw new Error(`\`performanceInsightRetention\` for each instance must be the same as the one at cluster level, got instance '${instance.node.id}': ${instance.performanceInsightRetention}, cluster: ${this.performanceInsightRetention}`);
+    }
+
+    // If `performanceInsightEncryptionKey` is enabled on the cluster, the same parameter for each instance must be
+    // undefined or the same as the value at cluster level.
+    if (this.performanceInsightEncryptionKey && instance.performanceInsightEncryptionKey) {
+      const clusterKeyArn = this.performanceInsightEncryptionKey.keyArn;
+      const instanceKeyArn = instance.performanceInsightEncryptionKey.keyArn;
+      const compared = Token.compareStrings(clusterKeyArn, instanceKeyArn);
+
+      if (compared === TokenComparison.DIFFERENT) {
+        throw new Error(`\`performanceInsightEncryptionKey\` for each instance must be the same as the one at cluster level, got instance '${instance.node.id}': '${instance.performanceInsightEncryptionKey.keyArn}', cluster: '${this.performanceInsightEncryptionKey.keyArn}'`);
+      }
+      // Even if both of cluster and instance keys are unresolved, check if they are the same token.
+      if (compared === TokenComparison.BOTH_UNRESOLVED && clusterKeyArn !== instanceKeyArn) {
```

Comment:
Not sure about the validity of this condition 🤔
Given that we're already ignoring the `ONE_UNRESOLVED` case, we could maybe skip the check for `BOTH_UNRESOLVED` as well

Reply:
I was certainly wondering about this myself. However, I thought that if only one of the values is a token, it is impossible to compare them in any way, but if both are tokens, they can be compared by check whether both are the same token.

I thought that excluding this case would diminish the validation as it would mean skipping the most common cases, such as the following. What do you think? (I still wonder...)

```ts
new DatabaseCluster(stack, 'Database', {
  engine: DatabaseClusterEngine.AURORA,
  vpc,
  performanceInsightEncryptionKey: new kms.Key(stack, 'Key1'),
  writer: ClusterInstance.provisioned('writer', {
    performanceInsightEncryptionKey: new kms.Key(stack, 'Key2'),
  }),
});
```